src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkRuleOrCheckpoint.kt

```diff
@@ -5,4 +5,17 @@ import com.jetbrains.python.psi.PyTypedElement
 
 interface SmkRuleOrCheckpoint : SmkRuleLike<SmkRuleOrCheckpointArgsSection>, PyTypedElement {
     fun collectWildcards(): List<Pair<PsiElement, String>>
+
+    /**
+     * Snakemake wildcards can be defined by 3 sections.
+     * Defining section priority is: output > log > benchmark
+     * @return Section that defines wildcards or null
+     *         if there are no such sections
+     */
+    fun getWildcardDefiningSection() =
+        statementList.statements.filter {
+            it.name in SmkRuleOrCheckpointArgsSection.KEYWORDS_GENERATING_WILDCARDS
+        }.minBy { pyStatement ->
+            SmkRuleOrCheckpointArgsSection.KEYWORDS_GENERATING_WILDCARDS.indexOf(pyStatement.name)
+        } as SmkRuleOrCheckpointArgsSection?
```

Comment:
This could result in future ClassCastException if `it.name` is some other psi element. I suggest
```
    fun getWildcardDefiningSection() = statementList.statements.asSequence()
            .filterIsInstance(SmkRuleOrCheckpointArgsSection::class.java)
            .filter { it.sectionKeyword in SmkRuleOrCheckpointArgsSection.KEYWORDS_GENERATING_WILDCARDS }
            .minBy { section ->
                SmkRuleOrCheckpointArgsSection.KEYWORDS_GENERATING_WILDCARDS.indexOf(section.sectionKeyword)
            }
```